frontend/src/features/env/SwitchEnvFeedbackModal.tsx

```diff
@@ -146,85 +148,33 @@ export const SwitchEnvFeedbackModal = ({
-                            },
-                            deps: [FEEDBACK_OTHERS_INPUT_NAME],
-                          })}
-                          value={option}
-                          key={option}
-                          tabIndex={1}
-                        >
-                          {option}
-                        </Radio>
-                      ))}
-                      {ADMIN_RADIO_OPTIONS.map((option) => (
-                        <Radio
-                          {...register('switchReason', {
-                            required: {
-                              value: true,
-                              message: 'This field is required',
-                            },
-                            deps: [FEEDBACK_OTHERS_INPUT_NAME],
-                          })}
-                          value={option}
-                          key={option}
-                        >
-                          {option}
-                        </Radio>
-                      ))}
-                      <Radio.OthersWrapper
-                        {...register('switchReason', {
-                          required: {
-                            value: true,
-                            message: 'This field is required',
-                          },
-                          deps: [FEEDBACK_OTHERS_INPUT_NAME],
-                        })}
-                        value={othersInputValue}
-                      >
-                        <FormControl>
-                          <OthersInput
-                            aria-label='"Other" response'
-                            {...register(FEEDBACK_OTHERS_INPUT_NAME, {
-                              validate: (value) => {
-                                return (
-                                  getValues('switchReason') !==
-                                    othersInputValue ||
-                                  !!value ||
-                                  'Please specify a value for the "Others" option'
-                                )
-                              },
-                            })}
-                          />
-                        </FormControl>
-                      </Radio.OthersWrapper>
-                    </Radio.RadioGroup>
+                  <FormControl isRequired isInvalid={!isEmpty(errors)}>
+                    <FormLabel>How was your experience using FormSG?</FormLabel>
+                    <Controller
+                      rules={{ required: 'This field is required' }}
+                      control={control}
+                      name={'rating'}
+                      render={({ field: { value, onChange, ...rest } }) => (
+                        <Rating
+                          numberOfRatings={5}
+                          variant={'star'}
+                          helperText={'1: Poor, 5: Excellent'}
+                          value={Number(value)}
+                          onChange={(val) => onChange(val?.toString())}
```

Comment:
qn: why are we casting this to string?

Reply:
the type of the `Rating` component and the `onChange` provided by the controller is different in that one allows undefined while the other does not, so we need it to get the types to work out 😓

Reply:
also, public form `RatingField` does the same thing oop hahaha